layer/contracts/definitions.py

```diff
@@ -128,11 +104,11 @@ def runner_function(self) -> Any:
         if self.asset_type == AssetType.DATASET:
             from layer.executables.entrypoint.dataset import DatasetRunner
 
-            return DatasetRunner(self)
+            return DatasetRunner(self, context.get_run_context())
```

Comment:
same question here, why not ask for context in 
```python
def runner_function(self, context: Context) -> Any:
```
?

it will become hard to unit test because we're hiding dependency injection when we're invoking context outside of constructors

Reply:
agreed, that can be passed in. will update!